### ISSUE TITLE
Parser does accept dot in type

### ISSUE THREAD

**Issue by StefanLobbenmeier** [exigow/intellij-gdscript#108]
I get this error

> ')', ',', ':=', '=', LINE_BREAK or '[' expected, got '.'

This should be enough code to reproduce:

A.gd
```gdscript 
class SubA:
```

B.gd
```gdscript 
var A = preload("A.gd")

func a(par: A.SubA) ->  A.SubA:
```

Screenshot:
![image](https://user-images.githubusercontent.com/36509607/146458284-cbf92ad3-b38b-4d71-bfe3-a1cf68a80b63.png)

Another screenshot after removing .SubA from the parameter:
![image](https://user-images.githubusercontent.com/36509607/146458370-11f5e79c-724a-46c7-b542-d5ae0b3ce936.png)

**Comment by StefanLobbenmeier** [comment]
Also loaded them in godot and fixed them to be accepted by godot linter:

A.gd
```gdscript 
class SubA:
	func init():
		print("")
```

B.gd
```gdscript 
const A = preload("A.gd")

func a(par: A.SubA) ->  A.SubA:
	return A.SubA.new()
```

Error is still the same though:
![image](https://user-images.githubusercontent.com/36509607/146458941-1a5a421c-6722-400b-83eb-6288619cd8e7.png)

No error in Godot:
![image](https://user-images.githubusercontent.com/36509607/146458958-2cd01360-5e2b-4981-a5b1-db2de7bd73de.png)

**Comment by exigow** [comment]
Thanks @StefanLobbenmeier for description. Fix is included in latest version (0.34.6).
